### ISSUE TITLE
When have multi selects some doesn't works

### ISSUE THREAD

**Issue by nunowar** [selectize/selectize.js#756]
I use this plugin and yesterday I update from v0.11.2 to v0.12.0, and some selects crashed.

They have content but appears without content after rendering.

I use last version of jQuery.

$('.select').selectize();

![captura de ecra 2015-04-3 as 20 42 49](https://cloud.githubusercontent.com/assets/1176835/6987718/37e3ff3c-da42-11e4-8e32-f392657e7c09.png)

**Comment by brianreavis** [comment]
Need more info – what's the HTML of the selects look like?

**Comment by nunowar** [comment]
http://pastebin.com/dQ7yS0id

The select with name="mes" doesn't works and name="uti_sexo" doesn't works too.

**Comment by Alex-Sokolov** [comment]
Same issue. 
Generated HTML:

``` html
<div class="row">
    <div class="column column-w" data-label="Фамилия">
        <select class="selectColumns" multiple="" name="Pages[0].ColumnsMergeSettings[0].ValueMergeRules">
            <option value="">Выберите колонки документа...</option>
            <option selected value="1">
                Колонка #1
            </option>
            <option value="2">
                Колонка #2
            </option>
            <option value="3">
                Колонка #3
            </option>
            <option value="4">
                Колонка #4
            </option>
            <option value="5">
                Колонка #5
            </option>
        </select>
    </div>
    <div class="column column-w" data-label="Имя">
        <select class="selectColumns" multiple="" name="Pages[0].ColumnsMergeSettings[1].ValueMergeRules">
            <option value="">Выберите колонки документа...</option>
            <option value="1">
                Колонка #1
            </option>
            <option selected value="2">
                Колонка #2
            </option>
            <option value="3">
                Колонка #3
            </option>
            <option value="4">
                Колонка #4
            </option>
            <option value="5">
                Колонка #5
            </option>
        </select>
    </div>
    <div class="column column-w" data-label="Отчество">
        <select class="selectColumns" multiple="" name="Pages[0].ColumnsMergeSettings[2].ValueMergeRules">
            <option value="">Выберите колонки документа...</option>
            <option value="1">
                Колонка #1
            </option>
            <option value="2">
                Колонка #2
            </option>
            <option selected value="3">
                Колонка #3
            </option>
            <option value="4">
                Колонка #4
            </option>
            <option value="5">
                Колонка #5
            </option>
        </select>
    </div>
    <div class="column column-w" data-label="Должность">
        <select class="selectColumns" multiple="" name="Pages[0].ColumnsMergeSettings[3].ValueMergeRules">
            <option value="">Выберите колонки документа...</option>
            <option value="1">
                Колонка #1
            </option>
            <option value="2">
                Колонка #2
            </option>
            <option value="3">
                Колонка #3
            </option>
            <option selected value="4">
                Колонка #4
            </option>
            <option value="5">
                Колонка #5
            </option>
        </option>
    </select>
</div>
</div>
```

Without selectize.js
![1](https://cloud.githubusercontent.com/assets/4497128/7004932/3deff792-dc7b-11e4-95ea-46e21ad8f91b.PNG)

When init with code:

``` js
$('.selectColumns').selectize({
    plugins: {
        'remove_button': { title: 'Удалить выбранный вариант' }
    }
});
```

![2](https://cloud.githubusercontent.com/assets/4497128/7004940/56f23034-dc7b-11e4-9687-6b8c3ed75588.PNG)

But I found workaround! With code:

``` js
$('.selectColumns').each(function () {
    $(this).selectize({
        plugins: {
            'remove_button': { title: 'Удалить выбранный вариант' }
        }
    });
});
```

All work as need
![3](https://cloud.githubusercontent.com/assets/4497128/7004967/9e19b7f2-dc7b-11e4-9772-a1cae38f1dac.PNG)
Using selectize.js (v0.12.0)

P.S.: Same problem when need to clear all selects in form. Can't clear all selectized inputs with one line

``` js
$('.selectColumns').each(function () {
    this.selectize.clear();
});
```

**Comment by robertsass** [comment]
I had the same issue and I really have no clue, what was causing it.
I used Bootstrap Tabs and had three identical selects: on one tab one select, on the other two (one of them in a modal dialog). The last one in the modal was empty, while another different select in that modal worked fine.

Initializing them by using .each() worked for me too.

**Comment by bb1** [comment]
I got the same problem when initializing large amounts of selects at the same time with a lot of options (up to 100,000 options in one select). But I FOUND A HOTFIX:

Before I used to call 
```javascript
$('.selectize').selectize( options );
```
now I use this:
```javascript
$('.selectize').each(function(){
    $(this).selectize( options );
}
```
And it works just fine. I guess you got a problem with some variables that are not cleared between multiple selects.

**Comment by bb1** [comment]
Had the same problem but the workaround that Alex-Sokolov posted worked just fine.
I got a large amount of selects with a lot of options (~100000). Maybe there is a buffer that just runs full.
I'm using FireFox btw.

**Comment by anaphel** [comment]
Same issue for me, with just a few options.
Had to do a loop on each select to make it work.

**Comment by jensljungblad** [comment]
+1

**Comment by johanhalse** [comment]
It seems the `optionsMap` variable isn't being reset between selectize initializations. When using $('.foo').selectize() you only get one instance of Selectize which re-uses the optionsMap var between select elements.

A quick fix is adding `optionsMap = {};` to the `init_select` function here: https://github.com/brianreavis/selectize.js/blob/master/src/selectize.jquery.js#L47

I don't know the code well enough to know if that breaks anything else though.

EDIT:
Looks like this is already fixed in master? https://github.com/brianreavis/selectize.js/commit/f6c9ac9bfeaacff4099588e2a52c3cd2694e89ad
But looking at the standalone dist js (which is what I'm using) the fix isn't there:
https://github.com/brianreavis/selectize.js/blob/master/dist/js/standalone/selectize.js#L3292

**Comment by brianreavis** [comment]
@johanhalse (and others) This was fixed by #717, which is now part of [0.12.1](https://github.com/brianreavis/selectize.js/releases/tag/v0.12.1)

**Comment by Alex-Sokolov** [comment]
@brianreavis Thanks! Work now in 0.12.1

But is it way to clear all selectized inputs without .each()?

``` js
$('.selectColumns').each(function () {
    this.selectize.clear();
});
```
